Create training-textclassification-activelearning-with-GPU.ipynb

MoritzLaurer opened the pull request:
uploading a new tutorial for running argilla in a google colab with a GPU, following the discussion here: https://github.com/argilla-io/argilla/issues/1901

MoritzLaurer committed:
Create training-textclassification-activelearning-with-GPU.ipynb

uploading a new tutorial for running argilla in a google colab with a GPU, following the discussion here: https://github.com/argilla-io/argilla/issues/1901

davidberenstein1957 commented on docs/_source/tutorials/notebooks/training-textclassification-activelearning-with-GPU.ipynb:
```diff
@@ -0,0 +1,695 @@
+{
```
Could you rename the tutorial to 
docs/_source/tutorials/notebooks/training-textclassification-colab-activelearning.ipynb ?

davidberenstein1957 replied:
This is according to our structure. mlops-nlp-package-technique

davidberenstein1957 replied:
Also, the tutorial should be added to the readme.md in 
`docs/_source/libraries/colab.md`
`docs/_source/techniques/active_learning.md`
`docs/_source/steps/training.md`
`docs/_source/tasks/textclassification.md`

davidberenstein1957 replied:
do you think it would be possible to add some pictures to make it a bit more visual?🤓

davidberenstein1957 commented on docs/_source/tutorials/notebooks/training-textclassification-activelearning-with-GPU.ipynb:
```diff
@@ -0,0 +1,695 @@
+    {
+      "cell_type": "markdown",
+      "source": [
+        "\n",
```
Could you limit the new line usage throughout the cells within the tutorial? Within VScode there also is the option to format the notebook via a right click.

davidberenstein1957 commented on docs/_source/tutorials/notebooks/training-textclassification-activelearning-with-GPU.ipynb:
```diff
@@ -0,0 +1,695 @@
+        "\n",
+        "1. Option with Colab Pro:\n",
+        " Open the Colab Pro terminal (button to the bottom left) and type in the terminal: \"python -m argilla\"\n",
+        "2. Option without Colab Pro: \n",
```
A newline is not started at 2.

davidberenstein1957 commented on docs/_source/tutorials/notebooks/training-textclassification-activelearning-with-GPU.ipynb:
```diff
@@ -0,0 +1,695 @@
+    {
+      "cell_type": "markdown",
+      "source": [
+        "In the Colab interface, you can choose a CPU (for initial testing) or a GPU (for an efficient active learning loop) by clicking Runtim > Change runtime type > Hardware accelerator in the menu in the top left. Once you have chosen your hardware, install the required packages. "
```
small typo Runtime

davidberenstein1957 commented on docs/_source/tutorials/notebooks/training-textclassification-activelearning-with-GPU.ipynb:
```diff
@@ -0,0 +1,695 @@
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Elastic Search is a requirement for using argilla. The [docker installation](https://docs.argilla.io/en/latest/getting_started/quickstart.html) of Elastic Search recommended by argilla does not work in Google Colab as [Colab does not support docker](https://github.com/googlecolab/colabtools/issues/299). Elastic Search therefore needs to be installed 'manually' with the follwing code."
```
small typo argilla->Argilla

davidberenstein1957 commented on docs/_source/tutorials/notebooks/training-textclassification-activelearning-with-GPU.ipynb:
```diff
@@ -0,0 +1,695 @@
+    {
+      "cell_type": "markdown",
+      "source": [
+        "You now need to start argilla localhost in separate terminal. We cannot just run \"!python -m argilla\" in a code cell on Colab, because the cell will run indefinitely and block us from running other cells. We therefore need to open a separate terminal to run argilla.\n",
```
Could you add code arguments within code brackets? `!python -m argilla`

davidberenstein1957 commented on docs/_source/tutorials/notebooks/training-textclassification-activelearning-with-GPU.ipynb:
```diff
@@ -0,0 +1,695 @@
+        "id": "837556c3"
+      },
+      "outputs": [],
+      "source": [
```
perfect

davidberenstein1957 reviewed (commented):
Hey, thanks for the great PR. I left some small remarks :)

MoritzLaurer committed:
Update and rename training-textclassification-activelearning-with-GPU.ipynb to training-textclassification-colab-activelearning.ipynb

made the requested changes on google colab (https://colab.research.google.com/drive/11oTWno3hzgJnip11EcgqEhdpbW1IX-lP?usp=sharing) and re-uploading the updated .ipynb file here
